lib/http/base.js

```diff
@@ -85,7 +85,14 @@ HttpClient.prototype.requestAsync = async function (options)
 {
   const requestOptions = prepareRequestOptions.call(this, options);
 
-  return axios.request(requestOptions);
+  let response = await axios.request(requestOptions);
+
+  if (Util.isString(response['data']) &&
```

Comment:
Hi @sfc-gh-ext-simba-lf - we have been refactoring and some questions arose from this part of code.
Specifically - did You ever noticed such case, that response['data'] is string, but 'content-type' headers are set to 'application/json' at the same time?

We expected axios to parse such data automatically (ad. e.g. https://masteringjs.io/tutorials/axios/json).

Reply:
Hi, the [tests](https://github.com/snowflakedb/snowflake-connector-nodejs/pull/672/files#diff-3399bc7071cb0f9829e2838ddaa9287958760a79f2a2f85d5074f3cadc8bcc5b) added with the PR can help repro that scenario. Or a simpler repro would be something like:
```
connection.execute({
	sqlText: "select 1;",
	asyncExec: true,
	complete: async function (err, stmt) {
		queryId = stmt.getQueryId();
		await connection.getQueryStatus(queryId);
	}
});
```

The `getQueryStatus` function receives a response in string that is parsed into JSON